pepper-apis/dsm-core/src/main/java/org/broadinstitute/dsm/model/kit/KitInitialScanUseCase.java

```diff
@@ -36,17 +37,8 @@ private Optional<ScanResult> updateKitRequest(String hruid, String kit) {
             Optional<KitRequestShipping> kitRequestWithPrefix =
                     kitList.stream().filter(k -> StringUtils.isNotBlank(k.getKitLabelPrefix())).findFirst();
             for (KitRequestShipping kitRequest : kitList) {
-                if (StringUtils.isNotBlank(kitRequest.getKitLabelPrefix()) && kit.startsWith(kitRequest.getKitLabelPrefix())) {
-                    // blood kit
-                    setKitInformation(kitRequest, kit, hruid);
-                    return kitDao.updateKitLabel(kitRequest);
-                } else if (StringUtils.isBlank(kitRequest.getKitLabelPrefix())
-                        && kitRequestWithPrefix.isPresent() && !kit.startsWith(kitRequestWithPrefix.get().getKitLabelPrefix())) {
-                    // saliva kit and blood kit in the queue -> so check that kit does not start with prefix
-                    setKitInformation(kitRequest, kit, hruid);
-                    return kitDao.updateKitLabel(kitRequest);
-                } else if (StringUtils.isBlank(kitRequest.getKitLabelPrefix()) && !kitRequestWithPrefix.isPresent()) {
-                    // saliva kit and no blood kit in the queue -> DSM doesn't know prefix then
+                if (isBloodKit(kitRequest, kit) || salivaAndBloodKitInQueue(kitRequest, kit, kitRequestWithPrefix)
+                        || salivaAndNoBloodInQueue(kitRequest, kitRequestWithPrefix)) {
```

Comment:
some refactoring here. I moved the condition checks into methods and gave them a proper name. Since in case of all of these condition the same thing was happening I just call all of them in an OR clause now